awscrt/http.py

```diff
@@ -144,6 +145,82 @@ def on_incoming_headers_received(headers, response_code, has_body):
 
         return request
 
+
+'''
+TODO: Server Connection class,and configure the server connection.
+'''
+
+
+class HttpServer(object):
+    """
+    Represents an Http server. Everything in this class is non-blocking.
+    """
+    __slots__ = ('_bootstrap', '_tls_connection_options', '_on_incoming_connection',
+                 '_on_destroy_complete', '_native_handle', 'destroy_complete')
+
+    # don't call me, I'm private
+    def __init__(self, bootstrap, on_incoming_connection, on_destroy_complete, tls_connection_options):
+        assert isinstance(bootstrap, ServerBootstrap)
+        assert tls_connection_options is None or isinstance(
+            tls_connection_options, TlsConnectionOptions)
+
+        for slot in self.__slots__:
+            setattr(self, slot, None)
+
+        self._bootstrap = bootstrap
+        self._tls_connection_options = tls_connection_options
+        self._on_incoming_connection = on_incoming_connection
+        self._on_destroy_complete = on_destroy_complete
+        self._native_handle = None
+        self.destroy_complete = Future()
+
+    
+    @staticmethod
+    def new_server(bootstrap, host_name, port, socket_options, on_incoming_connection, tls_connection_options=None):
+        """
+        Create a new server listener, binding to the host_name and port. 
+        When a new connection is received, the on_incoming_connection cb will be fired, a new ServerConnection obj will be created.
+        The aws_py_http_connection_configure_server need to be called from the callback to configure the ServerConnection 
+        @param socket_options: awscrt.io.SocketOptions for the server's listening socket. Required
+        @param on_incoming_connection: Callback with signature (connection: HttpConnection.native_handle, error_code: int) Required
+        @param bootstrap: awscrt.io.ServerBootstrap. Required
+        @param tls_connection_options: awscrt.io.TlsConnectionOptions, for TLS connection
+        """
+        assert tls_connection_options is None or isinstance(tls_connection_options, TlsConnectionOptions)
+        assert host_name is not None
+        assert port is not None
+        assert socket_options is not None and isinstance(
+            socket_options, SocketOptions)
```

Comment:
don't need to check `not None`, because None can't be an instance of SocketOptions